include/oneapi/dpl/internal/dynamic_selection_impl/round_robin_policy_impl.h

```diff
@@ -66,7 +66,13 @@ namespace experimental{
 
     template<typename ...Args>
     selection_handle_t select(Args&&...) {
-      auto i = next_context_++ % num_contexts_;
+      size_t i=0;
+      if(next_context_==std::numeric_limits<universe_container_size_t>::max()){
+          next_context_=0;
```

Comment:
Should "next_context_=0" actually be "next_context_ = next_context_%num_contexts_ + 1"?  In the unlikely case that you actually overflow, you'd could skip some devices by always going back to the 1st device, even if you were part way through the list when you overflowed.

Reply:
I was so focused on the update logic, I didn't notice that the reset logic is not thread safe. You need to add thread safety to select.  When it was simple increment of an atomic, things were fine, but now that the logic is more complicated, it's not.

Reply:
Fixed